packages/fast-components-react-msft/src/carousel/carousel.tsx

```diff
@@ -0,0 +1,227 @@
+import * as React from "react";
+import Foundation, { HandledProps } from "@microsoft/fast-components-foundation-react";
+import {
+    CarouselHandledProps,
+    CarouselProps,
+    CarouselUnhandledProps,
+    Slide,
+    SlideTheme,
+} from "./carousel.props";
+import { Flipper, FlipperDirection } from "../flipper";
+import { Tabs } from "@microsoft/fast-components-react-base";
+import { TabsClassNameContract } from "@microsoft/fast-components-class-name-contracts-base";
+import { get } from "lodash-es";
+
+/**
+ * The carousel state interface
+ */
+export interface CarouselState {
+    /**
+     * Holds the active tab id to share with other controls
+     */
+    activeId: string;
+}
+
+class Carousel extends Foundation<
+    CarouselHandledProps,
+    CarouselUnhandledProps,
+    CarouselState
+> {
+    public static displayName: string = "Carousel";
+
+    protected handledProps: HandledProps<CarouselHandledProps> = {
+        managedClasses: void 0,
+        label: void 0,
+        activeId: void 0,
+        items: void 0,
+    };
+
+    private slides: Slide[];
+
+    /**
+     * Define constructor
+     */
+    constructor(props: CarouselProps) {
+        super(props);
+
+        if (this.props.items) {
+            this.state = {
+                activeId:
+                    typeof this.props.activeId === "string"
+                        ? this.props.activeId
+                        : this.props.items.length > 0
+                            ? get(this.props.items[0], "id")
+                            : "",
+            };
+
+            this.updateSlides();
+        }
+    }
+
+    /**
+     * React life-cycle method
+     */
+    public componentDidUpdate(): void {
+        if (this.slides !== this.props.items) {
+            this.updateSlides();
+        }
+    }
+
+    /**
+     * Renders the component
+     */
+    public render(): React.ReactElement<HTMLDivElement> {
+        return (
+            <div {...this.unhandledProps()} className={this.generateClassNames()}>
+                <Flipper
+                    direction={FlipperDirection.previous}
+                    onClick={this.handlePreviousInvoke}
+                    className={get(
+                        this.props,
+                        "managedClasses.carousel_flipperPrevious",
+                        ""
+                    )}
+                />
+                <Tabs
+                    label={this.props.label}
+                    activeId={this.state.activeId}
+                    onUpdate={this.handleUpdate}
+                    items={this.slides}
+                    managedClasses={this.generateTabsClassNames()}
+                />
+                <Flipper
+                    direction={FlipperDirection.next}
+                    onClick={this.handleNextInvoke}
```

Comment:
Regarding these flipper click behaviors; since these flippers are never exposed to screen readers, they would never be invoked on anything but a click. Since this is primarily click related, I would suggest naming as such.

Reply:
Sure

Reply:
Sure